crates/compiler/builtins/roc/Json.roc

```diff
@@ -797,35 +806,36 @@ expect
 
 # JSON STRING PRIMITIVE --------------------------------------------------------
 
-# TODO add support for 'null' decoding
-
 # Decode a Json string primitive into a RocStr
 #
 # Note that decodeStr does not handle leading whitespace, any whitespace must be
 # handled in json list or record decodin.
 decodeString = Decode.custom \bytes, @Json {} ->
-    { taken: strBytes, rest } = bytes |> takeJsonString
-
-    if List.isEmpty strBytes then
-        { result: Err TooShort, rest: bytes }
+    if List.startsWith bytes ['n', 'u', 'l', 'l'] then
+        { result: Ok "null", rest: List.drop bytes 4 }
     else
-        # Replace unicode escpapes with Roc equivalent
-        { outBytes: strBytesReplaced } =
-            replaceEscapedChars { inBytes: strBytes, outBytes: [] }
+        { taken: strBytes, rest } = takeJsonString bytes
 
-        # Try to parse RocStr from bytes
-        result =
-            strBytesReplaced
-            |> List.dropFirst # Remove starting quotation mark
-            |> List.dropLast # Remove ending quotation mark
-            |> Str.fromUtf8
-
-        when result is
-            Ok str ->
-                { result: Ok str, rest }
-
-            Err _ ->
-                { result: Err TooShort, rest: bytes }
+        if List.isEmpty strBytes then
+            { result: Err TooShort, rest: bytes }
+        else
+            # Replace unicode escpapes with Roc equivalent
+            { outBytes: strBytesReplaced } =
+                replaceEscapedChars { inBytes: strBytes, outBytes: [] }
+
+            # Try to parse RocStr from bytes
+            result =
+                strBytesReplaced
+                |> List.dropFirst # Remove starting quotation mark
+                |> List.dropLast # Remove ending quotation mark
```

Comment:
The `dropFirst` and `dropLast` could be combined into:

`|> List.sublist { start: 1, len: Num.subSaturated (List.len strBytesReplaced) 2 }`

...which I'd expect to run a bit faster! 😄

Reply:
Separately, we could drop the quotation marks before doing the escaping; that way the escaping wouldn't have to traverse the quotation marks (which of course don't need escaping!)